dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/main/java/org/apache/dolphinscheduler/plugin/task/sql/SqlTask.java

```diff
@@ -227,8 +230,17 @@ public void executeFuncAndSql(List<SqlBinds> mainStatementsBinds,
         try {
 
             // create connection
-            connection = DataSourceClientProvider.getInstance().getConnection(DbType.valueOf(sqlParameters.getType()),
-                    baseConnectionParam);
+            if (DbType.valueOf(sqlParameters.getType()) == KYUUBI) {
+                log.info("full jdbc url : {}, user : {} begin to build a connection",
+                        DataSourceUtils.getJdbcUrl(DbType.KYUUBI, baseConnectionParam), baseConnectionParam.getUser());
+                Class.forName(baseConnectionParam.getDriverClassName());
+                connection = DriverManager.getConnection(baseConnectionParam.getJdbcUrl(),
+                        baseConnectionParam.getUser(), baseConnectionParam.getPassword());
+            } else {
+                connection =
+                        DataSourceClientProvider.getInstance().getConnection(DbType.valueOf(sqlParameters.getType()),
+                                baseConnectionParam);
+            }
```

Comment:
I'm not sure if this is a good approach. If there are new data sources or task types in the future that need to share the same implementation.

Reply:
Yup. It's not  a good way to do it. Can we extract part of code into getConnection method ?